### PR TITLE
30 robust error handling

### LINKED ISSUE #30: Robust error handling
be able to handle backend errors/ data errors and etc

### PR DESCRIPTION
**Pull request by Rashmil-1999**
This PR adds an alert component that takes a message and a status code to be shown. I have only added it to 1 call, if it looks good then I can go ahead and try to add it in other places according to priority.

### CODE REVIEW

File: src/components/childComponents/AlertMessage.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import CloseIcon from "@material-ui/icons/Close";
+
+export const AlertMessage = ({ open, setOpen, message, alertSeverity, status }) => {
+	let subMessage = "";
+
+	if (status !== null && status === 401) {
+		subMessage = " Error: Unauthorized request. Please log in and try again.";
+	} else if (status !== null && status === 403) {
+		subMessage = " Error: Forbidden, you don't have enough privileges to run this execution.";
+	} else if (status !== null) {
```
Comment (longshuicy) [inline src/components/childComponents/AlertMessage.jsx]: Might as well just use "else" here. I'm assuming whenever this Alert Message component is set Open that means something is wrong.

File: src/components/childComponents/AlertMessage.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import Alert from "@material-ui/lab/Alert";
+import CloseIcon from "@material-ui/icons/Close";
+
+export const AlertMessage = ({ open, setOpen, message, alertSeverity, status }) => {
+	let subMessage = "";
+
+	if (status !== null && status === 401) {
+		subMessage = " Error: Unauthorized request. Please log in and try again.";
+	} else if (status !== null && status === 403) {
+		subMessage = " Error: Forbidden, you don't have enough privileges to run this execution.";
```
Comment (longshuicy) [inline src/components/childComponents/AlertMessage.jsx]: Can you just be more generalized saying `you don't have enough privileges` and leave out the `exeuction` part.

File: src/components/childComponents/RunAnalysis.jsx [hunk 3]
```diff
@@ -193,6 +205,15 @@ export default function ViewResult(props) {
...
 						}
 					})}
+
+					<AlertMessage
+						open={submitExecutionErrorOpen}
+						setOpen={submitExecutionErrorSetOpen}
+						message={submitExecutionErrorMessage}
+						alertSeverity={"error"}
+						status={submitExecutionErrorStatus}
+					/>
```
Comment (longshuicy) [inline src/components/childComponents/RunAnalysis.jsx]: If it's possible I would prefer putting this component on the parent component (for each route/page) or even the index root; so this code won't be duplicated else where. 
E.g. with the current setting you have to include <AlertMessage> Again on the "ViewAnalysis" Tab again.

**Review by longshuicy — CHANGES_REQUESTED** [review]

File: src/components/childComponents/AlertMessage.jsx (hunk 1 above)
Comment (Rashmil-1999) [reply src/components/childComponents/AlertMessage.jsx]: Yes, it seems fitting. I will update this.

File: src/components/childComponents/RunAnalysis.jsx (hunk 3 above)
Comment (Rashmil-1999) [reply src/components/childComponents/RunAnalysis.jsx]: Got it, makes sense, I have added it to the parent component and have only passed it to the run analysis tab for now.

**Comment by longshuicy** [comment]
hmm i tried to comment out the header when submitting to datawolf. I did get a 401 error but i didn't see any popup..

image

**Review by longshuicy — APPROVED** [review]

**Pull request merged**